### PR TITLE
Do not error on duplicate config keys

### LINKED ISSUE #401: Duplicate config errors in templates
I noticed a few templates issue errors about duplicate configuration.

For example, take a fresh `$ pulumi new kubernetes-gcp-yaml -y`. It works fine on 3.45.0; however, after an upgrade, I see:

```
  pulumi:pulumi:Stack (gkek8s-dev):
    error: an unhandled error occurred: 1 error occurred:
    	*
    -error: <nil>: found duplicate config gcp:project;
    -error: <nil>: found duplicate config gcp:region;
```

Is that because the template itself defines the configuration keys, as well as the dev file? Isn't that how default config works, though? The project-wide YAML file can define configuration keys/values, and individual stacks can override them?

Deleting those two configuration keys from the `Pulumi.yaml` file and leaving them as-is in `Pulumi.dev.yaml` fixes the issue.

I assume this could be a bug that belongs in pulumi/pulumi, but figured I'd start by filing here since it is affecting templates.

### PR DESCRIPTION
**Pull request by aq17**
Fixes #401

### CODE REVIEW

**Review by AaronFriel — CHANGES_REQUESTED** [review]

File: pkg/tests/example_transpile_test.go [hunk 1]
```diff
@@ -97,7 +97,8 @@ func TestGenerateExamples(t *testing.T) {
 			_, template, diags, err := codegen.LoadTemplate(exampleProjectDir)
 			require.NoError(t, err, "Loading project %v", dir)
 			require.False(t, diags.HasErrors(), diags.Error())
-			assert.Len(t, diags, 0, "Should have neither warnings nor errors")
+			// TODO: update examples to use `config` instead of `configuration`. For now, allow the "`configuration` field is deprecated" warning.
+			// assert.Len(t, diags, 0, "Should have neither warnings nor errors")
```
Comment (AaronFriel) [inline pkg/tests/example_transpile_test.go:101]: Can we change this to a loop to filter out diags containing `` "`configuration` field is deprecated" ``?

**Review by AaronFriel — APPROVED** [review]
LGTM!

**Comment by joeduffy** [comment]
@aq17 @AaronFriel Glad to see a test in here, just wasn't immediately obvious it was protecting against the duplicate config key issue that led to the problem. Does it cover that? If yes, great! If not, should we be adding a new test?

**Comment by AaronFriel** [comment]
It does & this PR removes the failure condition you found in #401. 

There are two follow ups here:
- Update our examples to remove the first warning on `up`. #404 
- Enhance `pulumi convert` to understand structured (typed) project level config, which will help us migrate our convert tests to use hierarchical config. #406 

```shell
❯ cd $(mktemp -d)

/tmp/tmp.qP70vtV5Ub 
❯ pulumi new kubernetes-gcp-yaml -y

Created project 'tmp.qP70vtV5Ub'

Please enter your desired stack name.
To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
Created stack 'dev'

Saved config

Your new project is ready to go! 

To perform an initial deployment, run `pulumi up`

/tmp/tmp.qP70vtV5Ub 
❯ pulumi preview
Previewing update (dev)

View Live: https://app.pulumi.com/friel/tmp.qP70vtV5Ub/dev/previews/edac5652-3dbc-4f18-bbfe-e7ca55a5c84f

     Type                           Name                Plan       Info
 +   pulumi:pulumi:Stack            tmp.qP70vtV5Ub-dev  create     1 warning; 1 message
 +   ├─ gcp:compute:Network         gke-network         create     
 +   ├─ gcp:compute:Subnetwork      gke-subnet          create     
 +   ├─ gcp:container:Cluster       gke-cluster         create     
 +   ├─ gcp:serviceAccount:Account  gke-nodepool-sa     create     
 +   └─ gcp:container:NodePool      gke-nodepool        create     


Diagnostics:
  pulumi:pulumi:Stack (tmp.qP70vtV5Ub-dev):
    Warning: Pulumi.yaml: root-level `configuration` field is deprecated; please use `config` instead.

    warning: unable to detect a global setting for GCP Project;
    Pulumi will rely on per-resource settings for this operation.
    Set the GCP Project by using:
        `pulumi config set gcp:project <project>`

Outputs:
    clusterId  : output<string>
    clusterName: "gke-cluster-2792fcd"
    kubeconfig : [secret]
    networkId  : output<string>
    networkName: "gke-network-f68e324"

**Pull request merged**